File: backend/app/routers/nutrition.py
from __future__ import annotations

from fastapi import APIRouter, Body, Query, Request

from app.services.nutrition_service import (
    close_nutrition_task,
    create_nutrition_task,
    list_nutrition_tasks,
    nutrition_ai_advice,
    nutrition_dashboard,
    nutrition_patient_detail,
)

router = APIRouter(prefix="/api/nutrition", tags=["nutrition"])


def _actor(request: Request) -> str:
    return request.headers.get("X-User-Id") or request.headers.get("x-operator-id") or "anonymous"


@router.get("/dashboard")
async def dashboard(
    dept: str | None = Query(None, description="科室名称"),
    dept_code: str | None = Query(None, description="科室代码"),
    deptCode: str | None = Query(None, description="兼容前端 deptCode 参数"),
    patient_scope: str = Query("in_dept", description="患者范围"),
    detail: bool = Query(False, description="是否计算详情级营养数据"),
):
    return {"code": 0, **await nutrition_dashboard(department=dept, dept_code=dept_code or deptCode, patient_scope=patient_scope, detail=detail)}


@router.get("/{patient_id}")
async def patient_detail(patient_id: str):
    return {"code": 0, **await nutrition_patient_detail(patient_id)}


@router.post("/{patient_id}/task")
async def create_task(patient_id: str, request: Request, payload: dict = Body(default={})):
    return {"code": 0, **await create_nutrition_task(patient_id, payload or {}, _actor(request))}


@router.get("/{patient_id}/tasks")
async def patient_tasks(patient_id: str):
    return {"code": 0, **await list_nutrition_tasks(patient_id)}


@router.post("/tasks/{task_id}/close")
async def close_task(task_id: str, request: Request, payload: dict = Body(default={})):
    return {"code": 0, **await close_nutrition_task(task_id, payload or {}, _actor(request))}


@router.post("/{patient_id}/ai-advice")
async def ai_advice(patient_id: str, refresh: bool = Query(False)):
    return {"code": 0, **await nutrition_ai_advice(patient_id, refresh=refresh)}

File: frontend/src/api/nutrition.ts
import axios from 'axios'

const api = axios.create({ baseURL: import.meta.env.VITE_API_BASE_URL ?? '', timeout: 60000 })

export type NutritionScopeParams = { dept?: string; dept_code?: string; patient_scope?: 'in_dept' | 'out_dept' | 'all' }
export type NutritionDashboardParams = NutritionScopeParams & { detail?: boolean }

export const getNutritionDashboard = (params?: NutritionDashboardParams) =>
  api.get('/api/nutrition/dashboard', { params })

export const getNutritionPatient = (patientId: string) =>
  api.get(`/api/nutrition/${patientId}`)

export const postNutritionTask = (patientId: string, payload: Record<string, any>) =>
  api.post(`/api/nutrition/${patientId}/task`, payload)

export const getNutritionTasks = (patientId: string) =>
  api.get(`/api/nutrition/${patientId}/tasks`)

export const closeNutritionTask = (taskId: string, payload?: Record<string, any>) =>
  api.post(`/api/nutrition/tasks/${taskId}/close`, payload || {})

export const postNutritionAiAdvice = (patientId: string, params?: { refresh?: boolean }) =>
  api.post(`/api/nutrition/${patientId}/ai-advice`, undefined, { params })
